sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/continuous/ContinuousExecution.scala

```diff
@@ -92,6 +104,8 @@ class ContinuousExecution(
     do {
       runContinuous(sparkSessionForStream)
     } while (state.updateAndGet(stateUpdate) == ACTIVE)
+
+    stopSources()
```

Comment:
with the new abstraction, we should only stop sources when the stream query ends, instead of each reconfiguration.

Reply:
As above, this looks like it's correctly implemented to me but we should keep an eye out for flakiness.

Reply:
thanks for reminding! Yea I'll keep an eye on it.